src/pages/profile/account.tsx

```diff
@@ -0,0 +1,199 @@
+        </Head>
+        <PageLayout>
+          <div className="w-full h-full flex flex-col justify-center items-center">
+            <div className="align-middle">404 User not found</div>
+            <div>
+              <button
+                className="text-neutral-400 rounded-md underline"
+                onClick={() => void router.push("/signup/")}
+              >
+                Go to Signup
+              </button>
+            </div>
+          </div>
+        </PageLayout>
+      </>
+    );
+    // router.push("/signup/");
+  }
+  if (isLoadingUserData || !userData) return <LoadingPage />;
+
+  const { name, image: imageURL, email } = userData;
+
+  const onUpdate = handleSubmit((formData) => {
+    const newData = { formData, ...userData };
+    if (newData != userData) {
+      updateUser({ ...userData, ...formData });
+      console.log("formData", formData);
+    }
+  });
+
+  return (
+    <>
+      <Head>
+        <title>Profile</title>
+      </Head>
+      <PageLayout>
+        <div className="relative h-48 bg-slate-600 border-b overscroll-y-scroll w-full border-x md:max-w-2xl">
+          <Image
+            src={imageURL ?? "https://picsum.photos/300/300"}
+            alt={`${name ?? ""}'s profile pic`}
+            width={128}
+            height={128}
+            className="absolute -mb-[64px] ml-4 rounded-md border-b border-2 bottom-0 left-0 bg-black"
+          />
+        </div>
+        <div className="h-[64px]"></div>
+        <div className="p-4">
+          <div className="text-2xl font-bold">{name}</div>
+          <div className="pb-4">{email}</div>
+        </div>
+        <div className="border-b border-slate-100"></div>
+        <div className="p-4 w-1/2">
+          <div className="font-bold pb-2">
+            Account information
+            {isEditing ? (
+              <button
+                className="text-blue-400 rounded px-1 m-2"
+                onClick={() => setIsEditing(false)}
+              >
+                cancel
+              </button>
+            ) : (
+              <button
+                onClick={() => setIsEditing(true)}
+                className="text-blue-400 rounded px-1 m-2"
+              >
+                edit
+              </button>
+            )}
+          </div>
+          {isEditing && (
+            <>
+              <form className="flex flex-col items-start" onSubmit={void onUpdate}>
+                <label>name:</label>
+                <input
+                  className="text-slate-800"
+                  type="text"
+                  defaultValue={name}
+                  {...register("name")}
+                />
+                <div className="p-1" />
+                <label>email:</label>
+                <input
+                  className="text-slate-800"
+                  type="email"
+                  defaultValue={email}
+                  {...register("email", {
+                    required: "minimum 6 char long",
```

Comment:
Take a look at the Zod resolver at https://github.com/react-hook-form/resolvers

Reply:
thanks that helps